lib/postgres.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Library made to simplify accessing / connecting to postgres databases,
+ * and to cleanly handle when the pg module isn't installed.
+ * @author mia-pi-git
+ */
+
+// @ts-ignore in case module doesn't exist
+import type * as PG from 'pg';
+import type {SQLStatement} from 'sql-template-strings';
+import * as Streams from './streams';
+
+export class PostgresDatabase {
+	private pool: PG.Pool;
+	constructor(config = PostgresDatabase.getConfig()) {
+		try {
+			this.pool = new (require('pg').Pool)(config);
+		} catch (e) {
+			this.pool = null!;
+		}
+	}
+	async query(statement: string | SQLStatement, values?: any[]) {
+		if (!this.pool) {
+			throw new Error(`Attempting to use postgres without 'pg' installed`);
+		}
+		let result;
+		try {
+			result = await this.pool.query(statement, values);
+		} catch (e) {
+			// postgres won't give accurate stacks unless we do this
+			throw new Error(e.message);
+		}
+		return result?.rows || [];
+	}
+	static getConfig() {
+		let config: AnyObject = {};
+		try {
+			config = require('../config/config').usepostgres;
```

Comment:
Why can't we use the `Config` global or have configuration information passed in to the constructor? Both of those  sound significantly better than requiring a hardcoded configuration path.

Reply:
Passing a different configuration in the constructor. is supported. This is the default configuration, for ease of use.

Reply:
I thought `lib/` was supposed to be portable outside of PS? This makes it dependent on PS' directory structure, and I don't really think ease of use is impacted by having to call `new PostgresDatabase(Config.usepostgres)` instead of `new PostgresDatabase()`.

Reply:
Oh, forgot to reply to this. getConfig is the default, so yes it would be portable, since you you can pass in an object to override the default. This is really just a QoL.